src/main/java/frc/robot/systems/DriveFSMSystem.java

```diff
@@ -0,0 +1,693 @@
+				} else {
+					path5Points.add(new Pose2d(-AutoConstants.N_6, 0,
+						new Rotation2d(Math.toRadians(AutoConstants.DEG_180))));
+				}
+				return driveAlongPath(path5Points);
+			case PENDING:
+				return pause(AutoConstants.WAIT_TIME);
+			default:
+				return false;
+		}
+	}
+
+	/**
+	 * Resets the odometry to the specified pose.
+	 *
+	 * @param pose The pose to which to set the odometry.
+	 */
+	public void resetOdometry(Pose2d pose) {
+		odometry.resetPosition(
+			Rotation2d.fromDegrees(-gyro.getAngle()),
+				new SwerveModulePosition[] {
+					frontLeft.getPosition(),
+					frontRight.getPosition(),
+					rearLeft.getPosition(),
+					rearRight.getPosition()
+				},
+				pose);
+	}
+
+	/**
+	 * Update FSM based on new inputs. This function only calls the FSM state
+	 * specific handlers.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 */
+	public void update(TeleopInput input) {
+		odometry.update(Rotation2d.fromDegrees(-gyro.getAngle()),
+			new SwerveModulePosition[] {
+				frontLeft.getPosition(),
+				frontRight.getPosition(),
+				rearLeft.getPosition(),
+				rearRight.getPosition()});
+
+		SmartDashboard.putNumber("X Pos", getPose().getX());
+		SmartDashboard.putNumber("Y Pos", getPose().getY());
+		SmartDashboard.putNumber("Heading", getPose().getRotation().getDegrees());
+
+		if (input == null) {
+			return;
+		}
+		switch (currentState) {
+			case TELEOP_STATE:
+				drive(-MathUtil.applyDeadband((input.getControllerLeftJoystickY()
+					* Math.abs(input.getControllerLeftJoystickY())),
+					OIConstants.DRIVE_DEADBAND),
+					-MathUtil.applyDeadband((input.getControllerLeftJoystickX()
+					* Math.abs(input.getControllerLeftJoystickX())),
+					OIConstants.DRIVE_DEADBAND),
+					-MathUtil.applyDeadband(input.getControllerRightJoystickX(),
+					OIConstants.DRIVE_DEADBAND), true, true);
+				if (input.isBackButtonPressed()) {
+					gyro.reset();
+					resetOdometry(new Pose2d(new Translation2d(getPose().getX(), getPose().getY()),
+						new Rotation2d(0)));
+				}
+				break;
+
+			case ALIGN_TO_OBJECT_STATE:
+				//double dist = rpi.getConeDistance();
+				//double angle = rpi.getConeYaw();
+				//boolean canSee = (dist == -1 || angle == -1);
+				//SmartDashboard.putBoolean("Can See Object", canSee);
+				//SmartDashboard.putNumber("Distance to Object", dist);
+				//driveUntilObject(dist, angle);
+				break;
+
+			case ALIGN_TO_TAG_STATE:
+				// double xdist =  rpi.getAprilTagX(1);
+				// double ydist = rpi.getAprilTagY(1);
+				// double angle = rpi.getAprilTagYaw(1);
+				// boolean canSee = (xdist == AutoConstants.UNABLE_TO_SEE_TAG_CONSTANT
+				// 	|| angle == AutoConstants.UNABLE_TO_SEE_TAG_CONSTANT
+				// 	|| ydist == AutoConstants.UNABLE_TO_SEE_TAG_CONSTANT);
+				// SmartDashboard.putBoolean("Can See Tag", canSee);
+				// System.out.println("x tag: " + rpi.getAprilTagX(1));
+				// driveToTag(0, xdist, 0);
```

Comment:
For now, just remove all these comments for both the CV states. I'll add the tested code later in the Automation PR